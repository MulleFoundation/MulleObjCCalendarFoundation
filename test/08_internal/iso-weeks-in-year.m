#ifdef __MULLE_OBJC__
# import <MulleObjCCalendarFoundation/MulleObjCCalendarFoundation.h>
#else
# import <Foundation/Foundation.h>
#endif


int   main( void)
{
   NSCalendar                         *calendar;
   NSDateComponents                   *components;
   NSDate                             *date;
   NSInteger                          day;
   NSInteger                          year;
   struct MulleExtendedTimeInterval   extended;

   calendar = [NSCalendar calendarWithIdentifier:NSGregorianCalendar];
   [calendar setTimeZone:[NSTimeZone timeZoneWithAbbreviation:@"GMT"]];
   [calendar setFirstWeekday:2];
   [calendar setMinimumDaysInFirstWeek:4];

   for( year = 1600; year <= 2000; year++)
   {
      components = [[NSDateComponents new] autorelease];
      [components setYear:year];

      date       = [calendar dateFromComponents:components];
      components = [calendar components:~0
                               fromDate:date];
      MulleExtendedTimeIntervalInit( &extended, [date timeIntervalSinceReferenceDate]);

      printf( "%ld.%ld.%ld ",
            (long) [components day], (long) [components month], (long) [components year]);

      printf( " weeks: %ld\n",
               [calendar mulleISONumberOfWeeksInYearFromExtendedTimeInterval:&extended]);
   }
   return( 0);
}