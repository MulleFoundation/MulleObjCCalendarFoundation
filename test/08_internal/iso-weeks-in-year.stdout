1.1.1600  weeks: 52
1.1.1601  weeks: 52
1.1.1602  weeks: 52
1.1.1603  weeks: 52
1.1.1604  weeks: 53
1.1.1605  weeks: 52
1.1.1606  weeks: 52
1.1.1607  weeks: 52
1.1.1608  weeks: 52
1.1.1609  weeks: 53
1.1.1610  weeks: 52
1.1.1611  weeks: 52
1.1.1612  weeks: 52
1.1.1613  weeks: 52
1.1.1614  weeks: 52
1.1.1615  weeks: 53
1.1.1616  weeks: 52
1.1.1617  weeks: 52
1.1.1618  weeks: 52
1.1.1619  weeks: 52
1.1.1620  weeks: 53
1.1.1621  weeks: 52
1.1.1622  weeks: 52
1.1.1623  weeks: 52
1.1.1624  weeks: 52
1.1.1625  weeks: 52
1.1.1626  weeks: 53
1.1.1627  weeks: 52
1.1.1628  weeks: 52
1.1.1629  weeks: 52
1.1.1630  weeks: 52
1.1.1631  weeks: 52
1.1.1632  weeks: 53
1.1.1633  weeks: 52
1.1.1634  weeks: 52
1.1.1635  weeks: 52
1.1.1636  weeks: 52
1.1.1637  weeks: 53
1.1.1638  weeks: 52
1.1.1639  weeks: 52
1.1.1640  weeks: 52
1.1.1641  weeks: 52
1.1.1642  weeks: 52
1.1.1643  weeks: 53
1.1.1644  weeks: 52
1.1.1645  weeks: 52
1.1.1646  weeks: 52
1.1.1647  weeks: 52
1.1.1648  weeks: 53
1.1.1649  weeks: 52
1.1.1650  weeks: 52
1.1.1651  weeks: 52
1.1.1652  weeks: 52
1.1.1653  weeks: 52
1.1.1654  weeks: 53
1.1.1655  weeks: 52
1.1.1656  weeks: 52
1.1.1657  weeks: 52
1.1.1658  weeks: 52
1.1.1659  weeks: 52
1.1.1660  weeks: 53
1.1.1661  weeks: 52
1.1.1662  weeks: 52
1.1.1663  weeks: 52
1.1.1664  weeks: 52
1.1.1665  weeks: 53
1.1.1666  weeks: 52
1.1.1667  weeks: 52
1.1.1668  weeks: 52
1.1.1669  weeks: 52
1.1.1670  weeks: 52
1.1.1671  weeks: 53
1.1.1672  weeks: 52
1.1.1673  weeks: 52
1.1.1674  weeks: 52
1.1.1675  weeks: 52
1.1.1676  weeks: 53
1.1.1677  weeks: 52
1.1.1678  weeks: 52
1.1.1679  weeks: 52
1.1.1680  weeks: 52
1.1.1681  weeks: 52
1.1.1682  weeks: 53
1.1.1683  weeks: 52
1.1.1684  weeks: 52
1.1.1685  weeks: 52
1.1.1686  weeks: 52
1.1.1687  weeks: 52
1.1.1688  weeks: 53
1.1.1689  weeks: 52
1.1.1690  weeks: 52
1.1.1691  weeks: 52
1.1.1692  weeks: 52
1.1.1693  weeks: 53
1.1.1694  weeks: 52
1.1.1695  weeks: 52
1.1.1696  weeks: 52
1.1.1697  weeks: 52
1.1.1698  weeks: 52
1.1.1699  weeks: 53
1.1.1700  weeks: 52
1.1.1701  weeks: 52
1.1.1702  weeks: 52
1.1.1703  weeks: 52
1.1.1704  weeks: 52
1.1.1705  weeks: 53
1.1.1706  weeks: 52
1.1.1707  weeks: 52
1.1.1708  weeks: 52
1.1.1709  weeks: 52
1.1.1710  weeks: 52
1.1.1711  weeks: 53
1.1.1712  weeks: 52
1.1.1713  weeks: 52
1.1.1714  weeks: 52
1.1.1715  weeks: 52
1.1.1716  weeks: 53
1.1.1717  weeks: 52
1.1.1718  weeks: 52
1.1.1719  weeks: 52
1.1.1720  weeks: 52
1.1.1721  weeks: 52
1.1.1722  weeks: 53
1.1.1723  weeks: 52
1.1.1724  weeks: 52
1.1.1725  weeks: 52
1.1.1726  weeks: 52
1.1.1727  weeks: 52
1.1.1728  weeks: 53
1.1.1729  weeks: 52
1.1.1730  weeks: 52
1.1.1731  weeks: 52
1.1.1732  weeks: 52
1.1.1733  weeks: 53
1.1.1734  weeks: 52
1.1.1735  weeks: 52
1.1.1736  weeks: 52
1.1.1737  weeks: 52
1.1.1738  weeks: 52
1.1.1739  weeks: 53
1.1.1740  weeks: 52
1.1.1741  weeks: 52
1.1.1742  weeks: 52
1.1.1743  weeks: 52
1.1.1744  weeks: 53
1.1.1745  weeks: 52
1.1.1746  weeks: 52
1.1.1747  weeks: 52
1.1.1748  weeks: 52
1.1.1749  weeks: 52
1.1.1750  weeks: 53
1.1.1751  weeks: 52
1.1.1752  weeks: 52
1.1.1753  weeks: 52
1.1.1754  weeks: 52
1.1.1755  weeks: 52
1.1.1756  weeks: 53
1.1.1757  weeks: 52
1.1.1758  weeks: 52
1.1.1759  weeks: 52
1.1.1760  weeks: 52
1.1.1761  weeks: 53
1.1.1762  weeks: 52
1.1.1763  weeks: 52
1.1.1764  weeks: 52
1.1.1765  weeks: 52
1.1.1766  weeks: 52
1.1.1767  weeks: 53
1.1.1768  weeks: 52
1.1.1769  weeks: 52
1.1.1770  weeks: 52
1.1.1771  weeks: 52
1.1.1772  weeks: 53
1.1.1773  weeks: 52
1.1.1774  weeks: 52
1.1.1775  weeks: 52
1.1.1776  weeks: 52
1.1.1777  weeks: 52
1.1.1778  weeks: 53
1.1.1779  weeks: 52
1.1.1780  weeks: 52
1.1.1781  weeks: 52
1.1.1782  weeks: 52
1.1.1783  weeks: 52
1.1.1784  weeks: 53
1.1.1785  weeks: 52
1.1.1786  weeks: 52
1.1.1787  weeks: 52
1.1.1788  weeks: 52
1.1.1789  weeks: 53
1.1.1790  weeks: 52
1.1.1791  weeks: 52
1.1.1792  weeks: 52
1.1.1793  weeks: 52
1.1.1794  weeks: 52
1.1.1795  weeks: 53
1.1.1796  weeks: 52
1.1.1797  weeks: 52
1.1.1798  weeks: 52
1.1.1799  weeks: 52
1.1.1800  weeks: 52
1.1.1801  weeks: 53
1.1.1802  weeks: 52
1.1.1803  weeks: 52
1.1.1804  weeks: 52
1.1.1805  weeks: 52
1.1.1806  weeks: 52
1.1.1807  weeks: 53
1.1.1808  weeks: 52
1.1.1809  weeks: 52
1.1.1810  weeks: 52
1.1.1811  weeks: 52
1.1.1812  weeks: 53
1.1.1813  weeks: 52
1.1.1814  weeks: 52
1.1.1815  weeks: 52
1.1.1816  weeks: 52
1.1.1817  weeks: 52
1.1.1818  weeks: 53
1.1.1819  weeks: 52
1.1.1820  weeks: 52
1.1.1821  weeks: 52
1.1.1822  weeks: 52
1.1.1823  weeks: 52
1.1.1824  weeks: 53
1.1.1825  weeks: 52
1.1.1826  weeks: 52
1.1.1827  weeks: 52
1.1.1828  weeks: 52
1.1.1829  weeks: 53
1.1.1830  weeks: 52
1.1.1831  weeks: 52
1.1.1832  weeks: 52
1.1.1833  weeks: 52
1.1.1834  weeks: 52
1.1.1835  weeks: 53
1.1.1836  weeks: 52
1.1.1837  weeks: 52
1.1.1838  weeks: 52
1.1.1839  weeks: 52
1.1.1840  weeks: 53
1.1.1841  weeks: 52
1.1.1842  weeks: 52
1.1.1843  weeks: 52
1.1.1844  weeks: 52
1.1.1845  weeks: 52
1.1.1846  weeks: 53
1.1.1847  weeks: 52
1.1.1848  weeks: 52
1.1.1849  weeks: 52
1.1.1850  weeks: 52
1.1.1851  weeks: 52
1.1.1852  weeks: 53
1.1.1853  weeks: 52
1.1.1854  weeks: 52
1.1.1855  weeks: 52
1.1.1856  weeks: 52
1.1.1857  weeks: 53
1.1.1858  weeks: 52
1.1.1859  weeks: 52
1.1.1860  weeks: 52
1.1.1861  weeks: 52
1.1.1862  weeks: 52
1.1.1863  weeks: 53
1.1.1864  weeks: 52
1.1.1865  weeks: 52
1.1.1866  weeks: 52
1.1.1867  weeks: 52
1.1.1868  weeks: 53
1.1.1869  weeks: 52
1.1.1870  weeks: 52
1.1.1871  weeks: 52
1.1.1872  weeks: 52
1.1.1873  weeks: 52
1.1.1874  weeks: 53
1.1.1875  weeks: 52
1.1.1876  weeks: 52
1.1.1877  weeks: 52
1.1.1878  weeks: 52
1.1.1879  weeks: 52
1.1.1880  weeks: 53
1.1.1881  weeks: 52
1.1.1882  weeks: 52
1.1.1883  weeks: 52
1.1.1884  weeks: 52
1.1.1885  weeks: 53
1.1.1886  weeks: 52
1.1.1887  weeks: 52
1.1.1888  weeks: 52
1.1.1889  weeks: 52
1.1.1890  weeks: 52
1.1.1891  weeks: 53
1.1.1892  weeks: 52
1.1.1893  weeks: 52
1.1.1894  weeks: 52
1.1.1895  weeks: 52
1.1.1896  weeks: 53
1.1.1897  weeks: 52
1.1.1898  weeks: 52
1.1.1899  weeks: 52
1.1.1900  weeks: 52
1.1.1901  weeks: 52
1.1.1902  weeks: 52
1.1.1903  weeks: 53
1.1.1904  weeks: 52
1.1.1905  weeks: 52
1.1.1906  weeks: 52
1.1.1907  weeks: 52
1.1.1908  weeks: 53
1.1.1909  weeks: 52
1.1.1910  weeks: 52
1.1.1911  weeks: 52
1.1.1912  weeks: 52
1.1.1913  weeks: 52
1.1.1914  weeks: 53
1.1.1915  weeks: 52
1.1.1916  weeks: 52
1.1.1917  weeks: 52
1.1.1918  weeks: 52
1.1.1919  weeks: 52
1.1.1920  weeks: 53
1.1.1921  weeks: 52
1.1.1922  weeks: 52
1.1.1923  weeks: 52
1.1.1924  weeks: 52
1.1.1925  weeks: 53
1.1.1926  weeks: 52
1.1.1927  weeks: 52
1.1.1928  weeks: 52
1.1.1929  weeks: 52
1.1.1930  weeks: 52
1.1.1931  weeks: 53
1.1.1932  weeks: 52
1.1.1933  weeks: 52
1.1.1934  weeks: 52
1.1.1935  weeks: 52
1.1.1936  weeks: 53
1.1.1937  weeks: 52
1.1.1938  weeks: 52
1.1.1939  weeks: 52
1.1.1940  weeks: 52
1.1.1941  weeks: 52
1.1.1942  weeks: 53
1.1.1943  weeks: 52
1.1.1944  weeks: 52
1.1.1945  weeks: 52
1.1.1946  weeks: 52
1.1.1947  weeks: 52
1.1.1948  weeks: 53
1.1.1949  weeks: 52
1.1.1950  weeks: 52
1.1.1951  weeks: 52
1.1.1952  weeks: 52
1.1.1953  weeks: 53
1.1.1954  weeks: 52
1.1.1955  weeks: 52
1.1.1956  weeks: 52
1.1.1957  weeks: 52
1.1.1958  weeks: 52
1.1.1959  weeks: 53
1.1.1960  weeks: 52
1.1.1961  weeks: 52
1.1.1962  weeks: 52
1.1.1963  weeks: 52
1.1.1964  weeks: 53
1.1.1965  weeks: 52
1.1.1966  weeks: 52
1.1.1967  weeks: 52
1.1.1968  weeks: 52
1.1.1969  weeks: 52
1.1.1970  weeks: 53
1.1.1971  weeks: 52
1.1.1972  weeks: 52
1.1.1973  weeks: 52
1.1.1974  weeks: 52
1.1.1975  weeks: 52
1.1.1976  weeks: 53
1.1.1977  weeks: 52
1.1.1978  weeks: 52
1.1.1979  weeks: 52
1.1.1980  weeks: 52
1.1.1981  weeks: 53
1.1.1982  weeks: 52
1.1.1983  weeks: 52
1.1.1984  weeks: 52
1.1.1985  weeks: 52
1.1.1986  weeks: 52
1.1.1987  weeks: 53
1.1.1988  weeks: 52
1.1.1989  weeks: 52
1.1.1990  weeks: 52
1.1.1991  weeks: 52
1.1.1992  weeks: 53
1.1.1993  weeks: 52
1.1.1994  weeks: 52
1.1.1995  weeks: 52
1.1.1996  weeks: 52
1.1.1997  weeks: 52
1.1.1998  weeks: 53
1.1.1999  weeks: 52
1.1.2000  weeks: 52
